File: interfaces/frontend/src/lib/api/addresses.ts
import { post } from './client';
import type { components } from './schema';

type OkResponse = components['schemas']['OkResponse'];
type AddressReviewResponse = components['schemas']['AddressReviewResponse'];
type BatchUpdatedResponse = components['schemas']['BatchUpdatedResponse'];
type BatchCountryResponse = components['schemas']['BatchCountryResponse'];

export function review(
	addrId: number,
	body: Record<string, unknown>
): Promise<AddressReviewResponse> {
	return post<AddressReviewResponse>(`/api/addresses/${addrId}/review`, body);
}

export function batchReview(body: Record<string, unknown>): Promise<BatchUpdatedResponse> {
	return post<BatchUpdatedResponse>('/api/addresses/batch-review', body);
}

export function setCountry(addrId: number, body: Record<string, unknown>): Promise<OkResponse> {
	return post<OkResponse>(`/api/addresses/${addrId}/country`, body);
}

export function batchSetCountry(body: Record<string, unknown>): Promise<BatchCountryResponse> {
	return post<BatchCountryResponse>('/api/addresses/batch-country', body);
}

File: interfaces/api/models/addresses.py
"""Modèles Pydantic du router des adresses : corps des requêtes entrantes et réponses composées après mutation."""

from pydantic import BaseModel

from application.ports.api.addresses_queries import (
    AddressPublicationItem,
    AddressStructureSummary,
)

# ----- Corps des requêtes -----

# `is_confirmed` : True = confirmé, False = rejeté, None = retour à l'état non tranché.


class ReviewAction(BaseModel):
    structure_id: int
    is_confirmed: bool | None


class BatchReviewAction(BaseModel):
    address_ids: list[int]
    structure_id: int
    is_confirmed: bool | None


class SetCountry(BaseModel):
    countries: list[str] | None = None


class BatchSetCountry(BaseModel):
    country_code: str
    address_ids: list[int] | None = None
    search: str = ""
    has_country: bool | None = None
    country_code_filter: str = ""
    suggested_country: str = ""


# ----- Réponses composées par le router -----


class AddressPublicationsResponse(BaseModel):
    address_id: int
    raw_text: str
    publications: list[AddressPublicationItem]


class AddressReviewResponse(BaseModel):
    """Réponse de POST /api/addresses/{addr_id}/review."""

    id: int
    is_confirmed: bool | None
    is_detected: bool
    structures: list[AddressStructureSummary]


class BatchCountryResponse(BaseModel):
    """POST /api/addresses/batch-country : modifs directes + propagation."""

    updated: int
    propagated: int
